Firestore/Source/Core/FSTFirestoreClient.mm

```diff
@@ -107,42 +108,40 @@ + (instancetype)clientWithDatabaseInfo:(const DatabaseInfo &)databaseInfo
                    credentialsProvider:
                        (CredentialsProvider *)credentialsProvider  // no passing ownership
                           userExecutor:(std::unique_ptr<Executor>)userExecutor
-                   workerDispatchQueue:(FSTDispatchQueue *)workerDispatchQueue {
+                           workerQueue:(AsyncQueue *)workerQueue {
   return [[FSTFirestoreClient alloc] initWithDatabaseInfo:databaseInfo
                                            usePersistence:usePersistence
                                       credentialsProvider:credentialsProvider
                                              userExecutor:std::move(userExecutor)
-                                      workerDispatchQueue:workerDispatchQueue];
+                                              workerQueue:workerQueue];
 }
 
 - (instancetype)initWithDatabaseInfo:(const DatabaseInfo &)databaseInfo
                       usePersistence:(BOOL)usePersistence
                  credentialsProvider:
                      (CredentialsProvider *)credentialsProvider  // no passing ownership
                         userExecutor:(std::unique_ptr<Executor>)userExecutor
-                 workerDispatchQueue:(FSTDispatchQueue *)workerDispatchQueue {
+                         workerQueue:(AsyncQueue *)workerQueue {
   if (self = [super init]) {
     _databaseInfo = databaseInfo;
     _credentialsProvider = credentialsProvider;
     _userExecutor = std::move(userExecutor);
-    _workerDispatchQueue = workerDispatchQueue;
+    _workerQueue = workerQueue;
 
     auto userPromise = std::make_shared<std::promise<User>>();
     bool initialized = false;
 
     __weak __typeof__(self) weakSelf = self;
-    auto credentialChangeListener = [initialized, userPromise, weakSelf,
-                                     workerDispatchQueue](User user) mutable {
+    auto credentialChangeListener = [self, initialized, userPromise, weakSelf,
```

Comment:
Do you need both weakSelf and self?

Reply:
The original code uses `self` implicitly after checking `weakSelf`, I wanted to minimize the change.

Reply:
The original code only uses `self` via `__typeof__` (which is a nonstandard extension meaning something similar to `decltype`), which has a compile-time effect but doesn't result in a run-time capture.

Putting `self` in the explicit capture here retains the strong reference and defeats the purpose of passing this as a weak reference.

Objective-C does not support type inference so `__typeof__(self)` is the simplest way to say something like "same type just strong". Objective-C++ has auto so you could do this instead:

```c++
__strong auto strongSelf = weakSelf;
```

This is documented as working:
https://stackoverflow.com/questions/44690475/what-are-the-rules-for-deduced-type-by-auto-keyword-when-applied-in-objective